### PR TITLE
Add port to Breakers matcher

### PR DESCRIPTION
**Pull request by annaswims**
## Description of change
Breakers wasn't looking at the port when it was using matchers, therefore all services on "https://internal-dsva-vagov-prod-fwdproxyBLAH.amazonaws.com:PORT" were matching the first with a blank base_url path. 
## Testing done
integration testing for MVI

#### Applies to all PRs

- [ ] Appropriate logging
- [ ] Swagger docs have been updated, if applicable
- [ ] Provide link to originating GitHub issue, or connected to it via ZenHub
- [ ] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [ ] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

**Comment by annaswims** [comment]
I'd like to test that the regexes don't overlap, but they do. 

These configs all have the same base_path, but they're all mhv api calls, so it's not unreasonable for breakers to treat them similarly, I'm guessing. 
lib/bb/configuration.rb
lib/mhv_ac/configuration.rb 
lib/rx/configuration.rb

File: lib/common/client/configuration/base.rb [hunk 1]
```diff
@@ -47,11 +47,10 @@ def request_options
 
         def breakers_service
           return @service if defined?(@service)
-
-          path = URI.parse(base_path).path
-          host = URI.parse(base_path).host
+          base_uri = URI.parse(base_path)
```
Comment (annaswims) [inline lib/common/client/configuration/base.rb]: rubocop wanted one less line, hence the refactor bit. 🤷‍♀️

**Comment by annaswims** [comment]
I was struggling to test the matcher proc, but I think that the additional tests demonstrate that Breakers is functioning as expected after the change.  Additional testing strategies welcome.

File: spec/request/user_request_spec.rb [hunk 2]
```diff
@@ -159,21 +159,23 @@
           .and not_trigger_statsd_increment('api.external_http_request.MVI.skipped')
           .and not_trigger_statsd_increment('api.external_http_request.MVI.success')
       end
+      expect(MVI::Configuration.instance.breakers_service.latest_outage.start_time.to_i).to eq(start_time.to_i)
```
Comment (wyattwalter) [inline spec/request/user_request_spec.rb]: We could potentially test that another service on the same hostname didn't have an outage started at the same time as an integration test on the matcher.

**Comment by annaswims** [comment]
verified on stage that services were no longer all AppealsStatus http://prometheus-staging.vetsgov-internal:9090/prometheus/graph?g0.range_input=5m&g0.expr=sum(rate(api_external_http_request_duration_seconds_sum%5B1m%5D))+by+(service%2C+endpoint)&g0.tab=0

**Pull request merged**